File: 119-PascalTriangle2/pascalTriangle2.java
public class Solution {
    public List<Integer> getRow(int rowIndex) {

        Integer[] rslt = new Integer[rowIndex +1];
        rslt[0] = 1;
        for (int i = 1; i< rowIndex +1; i++) {
            //for factorial, use long to avoid possbile mistake!

            rslt[i] = (int)((long)rslt[i-1] * (rowIndex - i +1)/i);

        }
        return Arrays.asList(rslt);
        
    }
}

File: 119-PascalTriangle2/pascalTriangle2.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> pre(rowIndex+1);
        vector<int> cur(rowIndex+1);
        if (rowIndex<0)  return cur;

        pre[0]=1;
        cur[0]=1;
        for(int i=1;i<=rowIndex;i++){
            cur[0]=1;
            cur[i]=1;
            for(int j=1;j<i;j++){
                cur[j]=pre[j-1]+pre[j];
            }
            pre=cur;
        }
        return cur;
    }
};